bank_accounts.py

```diff
@@ -10,21 +20,83 @@
         bank_accounts = cursor.fetchall()
         cursor.close()
         conn.close()
         
-        return jsonify({
-            {'id': account['id'], 'user_id': account['user_id'], 'account_number': account['account_number'], 'balance': account['balance'],
-             'type': account['type'], 'currency': account['currency'], 'created_at': account['created_at'], 'status': account['status']}
+        accounts_list = [
+            {
+                'id': account['id'],
+                'user_id': account['user_id'],
+                'account_number': account['account_number'],
+                'balance': account['balance'],
+                'type': account['type'],
+                'currency': account['currency'],
+                'created_at': account['created_at'],
+                'status': account['status']
+            }
             for account in bank_accounts
-        }), 200
+        ]
+        
+        return jsonify(accounts_list), 200
         
     except Exception as e:
-        return jsonify({'error': 'Failed retrieving all bank accounts'}), 500
\ No newline at end of file
+        logging.debug("General Error occured.")
+        return jsonify({'error': 'Internal Error'}), 500
+    
+@bank_accounts_bp.route('/create_bank_account', methods=['POST'])
+def create_bank_account():
+    data = request.get_json()
+    user_id = data['user_id']
+    # required_fields = ['user_id', 'type', 'currency']
+    
+    # for field in required_fields:
+    #     if field not in data or not data[field].strip():
+    #         return jsonify({'error': f'the field {field} is missing'}), 400
+    try:
+        user_id = check_is_valid_user_id(str(data['user_id']))
+        if not user_id:
+            raise UserNotFoundError("Invalid user id.")
+
+        # Required params
+        account_id = str(uuid4())
+        account_number = str(uuid4().int)[:6]
+        encrypted_account_number = encrypt_account_number(account_number).decode('utf-8')
+        currency = 'USD'
+        balance = 0
+        account_type = 'CHECKINGS'
+        created_at = datetime.datetime.now()
+        status = 'ACTIVE'
+        
+        cursor, conn = get_db_connection()
+        if cursor is None or conn is None:
+            raise DatabaseConnectionError
+        cursor.execute("""
+            INSERT INTO bank_accounts (id, user_id, account_number, balance, type, currency, created_at, status) 
+            VALUES (%s, %s, %s, %s, %s, %s, %s, %s) RETURNING id;
+        """, (account_id, user_id, encrypted_account_number, balance, account_type, currency, str(created_at), status))
+        
+        conn.commit()
+        cursor.close()
+        conn.close()
+            
+    except DatabaseConnectionError as e:
+        return jsonify({'error': 'Internal Error'}), 500
+    except UserNotFoundError as e:
+        return jsonify({'error' : 'User not found'}), 404
+    except Exception as e:
+        return jsonify({'error', 'Internal Erorr'}), 500
```

Comment:
Kinda redundant here. What is the purpose of using special errors if you still need to iterate over them here? A nice thing to do would be something like "return e.Jsonify()" so it'll have the actual error generated for you without iterating over all of them yourself here.